File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/gdiplus/text/vb/gdiptext.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------



Imports System
Imports System.Windows.Forms
Imports System.IO
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Drawing.Text
Imports System.Reflection
Imports Microsoft.VisualBasic

Namespace Microsoft.Samples.WinForms.VB.GDIPlus.GDIPText

    Public Class TextSample
        Inherits System.Windows.Forms.Form

        Private backgroundBrush As System.Drawing.Brush
        Private textTextureBrush As System.Drawing.Brush
        Private titleFont As System.Drawing.Font
        Private textFont As System.Drawing.Font
        Private titleShadowBrush As System.Drawing.Brush
        Private flowedText1 As String = "I went down to the St James Infirmary," & ControlChars.CrLf & "Saw my baby there," & ControlChars.CrLf & "Stretched out on a long white table," & ControlChars.CrLf & "So sweet,so cold,so fair," & ControlChars.CrLf & "Let her go,let her go,God bless her," & ControlChars.CrLf & "Wherever she may be," & ControlChars.CrLf & "She can look this wide world over," & ControlChars.CrLf & "But she'll never find a sweet man like me," & ControlChars.CrLf & "When I die want you to dress me in straight lace shoes," & ControlChars.CrLf & "I wanna a boxback coat and a Stetson hat," & ControlChars.CrLf & "Put a twenty dollar gold piece on my watch chain," & ControlChars.CrLf & "So the boys'll know that I died standing up."
        Private flowedText2 As String = "the sky seems full when you're in the cradle" & ControlChars.CrLf & "the rain will fall and wash your dreams" & ControlChars.CrLf & "stars are stars and they shine so hard..." & ControlChars.CrLf & "now spit out the sky because its empty" & ControlChars.CrLf & "and hollow and all your dreams are hanging out to dry" & ControlChars.CrLf & "stars are stars and they shine so cold..." & ControlChars.CrLf & "once i like crying twice i like laughter" & ControlChars.CrLf & "come tell me what i'm after"
        Private japaneseFont As System.Drawing.Font
        Private japaneseText As New String(New Char() {ChrW(31169), ChrW(12398), ChrW(21517), ChrW(21069), ChrW(12399), ChrW(12463), ChrW(12522), ChrW(12473), ChrW(12391), ChrW(12377), ChrW(12290)})
        Private linearGradBrush As System.Drawing.Brush
        Private doJapaneseSample As Boolean = True

        Private serifFontFamily As FontFamily

        <STAThread()> Shared Sub Main()
            System.Windows.Forms.Application.Run(New TextSample())
        End Sub

        Public Sub New()

            MyBase.New()

            TextSample = Me

            InitializeComponent()

            serifFontFamily = New FontFamily(GenericFontFamilies.Serif)

            Me.SetStyle(ControlStyles.Opaque, True)

            '//Make sure we redraw on resize
            Me.SetStyle(ControlStyles.ResizeRedraw, True)

            Dim backgroundImage As Image

            '//Load the image to be used for the background from the exe's resource fork
            backgroundImage = New Bitmap(System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream("colorbars.jpg"))

            '//Now create the brush we are going to use to paint the background
            backgroundBrush = New TextureBrush(backgroundImage)

            '//Load the image to be used for the textured text from the exe's resource fork
            Dim textImage As Image = New Bitmap(System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream("marble.jpg"))
            textTextureBrush = New TextureBrush(textImage)

            '//Load the fonts we want to use
            Me.Font = New Font(serifFontFamily, 20)
            titleFont = New Font(serifFontFamily, 60)
            textFont = New Font(serifFontFamily, 11)

            '//Set up shadow brush - make it translucent
            titleShadowBrush = New SolidBrush(Color.FromArgb(70, Color.Black))

            '//Set up fonts and brushes for printing japanese text
            Try
                japaneseFont = New Font("MS Mincho", 36)
                linearGradBrush = New LinearGradientBrush(New Point(0, 0), New Point(0, 45), Color.Blue, Color.Red)
            Catch ex As Exception
                MessageBox.Show("The Japanese font MS Mincho needs be present to run the Japanese part of this sample" & ControlChars.CrLf & "" & ControlChars.CrLf & "" + ex.Message)
                doJapaneseSample = False
            End Try

        End Sub

        'Form overrides dispose to clean up the component list.
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub



#Region " Windows Form Designer generated code "

        'Required by the Windows Form Designer
        Private components As System.ComponentModel.Container

        Private WithEvents TextSample As System.Windows.Forms.Form

        'NOTE: The following procedure is required by the Windows Form Designer
        'It can be modified using the Windows Form Designer.
        'Do not modify it using the code editor.
        Private Sub InitializeComponent()
            Me.components = New System.ComponentModel.Container()
            Me.Size = New System.Drawing.Size(750, 500)
            Me.Text = "GDI+ Text Samples"
        End Sub

#End Region

        Protected Overrides Sub OnPaint(ByVal e As PaintEventArgs)
            Dim g As Graphics = e.Graphics

            g.SmoothingMode = SmoothingMode.AntiAlias

            '//Fill the background use the texture brush
            '//and then apply a white wash
            g.FillRectangle(backgroundBrush, ClientRectangle)
            g.FillRectangle(New SolidBrush(Color.FromArgb(180, Color.White)), ClientRectangle)

            '//Simple draw hello world
            g.DrawString("Hello World", Me.Font, New SolidBrush(Color.Black), 10, 10)

            '//Draw a textured string
            Dim titleText As String = "Graphics  Samples"
            g.DrawString(titleText, titleFont, titleShadowBrush, 15, 25)
            g.DrawString(titleText, titleFont, textTextureBrush, 10, 20)

            Dim textToDraw As String = "Hello Symetrical World"

            '//Use Measure string to display a string at the center of the window
            Dim windowCenter As Double = Me.DisplayRectangle.Width / 2
            Dim stringSize As SizeF = g.MeasureString(textToDraw, textFont)
            Dim startPos As Double = windowCenter - (stringSize.Width / 2)
            g.DrawString(textToDraw, textFont, New SolidBrush(Color.Red), CType(startPos, Single), 10)

            '//Now draw a string flowed into a rectangle
            Dim rectangle1 As RectangleF = New RectangleF(20, 150, 250, 300)
            g.FillRectangle(New SolidBrush(Color.Gainsboro), rectangle1)
            g.DrawString(flowedText1, textFont, New SolidBrush(Color.Blue), rectangle1)

            '//Draw more flowed text but this time center it
            Dim rectangle2 As RectangleF = New RectangleF(450, 150, 250, 300)
            g.FillRectangle(New SolidBrush(Color.Gainsboro), rectangle2)
            Dim format As StringFormat = New StringFormat()
            format.Alignment = StringAlignment.Center
            g.DrawString(flowedText2, textFont, New SolidBrush(Color.Blue), rectangle2, Format)

            '//Work out how many lines and characters we printed just now
            Dim characters As Integer = 0
            Dim lines As Integer = 0
            g.MeasureString(flowedText2, textFont, rectangle2.Size, format, characters, lines)
            Dim whatRenderedText As String = String.Format("We printed {0} characters and {1} lines", CType(characters, String), CType(lines, String))
            g.DrawString(whatRenderedText, textFont, New SolidBrush(Color.Black), 400, 440)

            '//If we have the Japanese language pack draw some text in Japanese
            '//Rotate it to make life truly exciting
            If (doJapaneseSample) Then
                g.RotateTransform(-30)
                g.TranslateTransform(-180, 300)
                g.DrawString(japaneseText, japaneseFont, linearGradBrush, 200, 140)
                g.ResetTransform()
            End If

        End Sub

    End Class

End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/gdiplus/text/cs/gdiptext.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.WinForms.Cs.GDIPlus.GDIPText {
    using System;
    using System.Windows.Forms;
    using System.IO;
    using System.Drawing;
    using System.Drawing.Drawing2D;
    using System.Drawing.Text;
    using System.Reflection;

    public class TextSample : Form {
        /// <summary>
        ///    Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components;

        private Brush backgroundBrush;
        private Brush textTextureBrush;
        private Font titleFont;
        private Font textFont;
        private Brush titleShadowBrush;
        private string flowedText1="I went down to the St James Infirmary,\nSaw my baby there,\nStretched out on a long white table,\nSo sweet,so cold,so fair,\nLet her go,let her go,God bless her,\nWherever she may be,\nShe can look this wide world over,\nBut she'll never find a sweet man like me,\nWhen I die want you to dress me in straight lace shoes,\nI wanna a boxback coat and a Stetson hat,\nPut a twenty dollar gold piece on my watch chain,\nSo the boys'll know that I died standing up.";
        private string flowedText2="the sky seems full when you're in the cradle\nthe rain will fall and wash your dreams\nstars are stars and they shine so hard...\nnow spit out the sky because its empty\nand hollow and all your dreams are hanging out to dry\nstars are stars and they shine so cold...\nonce i like crying twice i like laughter\ncome tell me what i'm after";
        private Font japaneseFont ;
        private string japaneseText = new string(new char[] {(char)31169, (char)12398, (char)21517, (char)21069, (char)12399, (char)12463, (char)12522, (char)12473, (char)12391, (char)12377, (char)12290});
        private Brush linearGradBrush;
        private bool doJapaneseSample=true;

        private FontFamily serifFontFamily;

        public TextSample() {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();

            serifFontFamily = new FontFamily (GenericFontFamilies.Serif);

            this.SetStyle(System.Windows.Forms.ControlStyles.Opaque, true);

            //Make sure form redraws when it is resized
            this.SetStyle(ControlStyles.ResizeRedraw,true);

            //Load the image to be used for the background from the exe's resource fork
            Image backgroundImage = new Bitmap(Assembly.GetExecutingAssembly().GetManifestResourceStream("colorbars.jpg"));

            //Now create the brush we are going to use to paint the background
            backgroundBrush = new TextureBrush(backgroundImage);

            //Load the image to be used for the textured text from the exe's resource fork
            Image textImage = new Bitmap(Assembly.GetExecutingAssembly().GetManifestResourceStream("marble.jpg"));
            textTextureBrush = new TextureBrush(textImage);

            //Load the fonts we want to use
            this.Font = new Font(serifFontFamily, 20);
            titleFont = new Font(serifFontFamily, 60);
            textFont = new Font(serifFontFamily, 11);

            //Set up shadow brush - make it translucent
            titleShadowBrush = new SolidBrush(Color.FromArgb(70, Color.Black));

            //Set up fonts and brushes for printing japanese text
            try {
                japaneseFont = new Font("MS Mincho", 36);
                linearGradBrush = new LinearGradientBrush(new Point(0, 0), new Point(0, 45), Color.Blue, Color.Red);
            } catch (Exception ex) {
                MessageBox.Show("The Japanese font MS Mincho needs be present to run the Japanese part of this sample\n\n" + ex.Message);
                doJapaneseSample = false;
            }

        }

        protected override void OnPaint(PaintEventArgs e) {
            Graphics g = e.Graphics;

            e.Graphics.SmoothingMode = SmoothingMode.AntiAlias;

            //Fill the background use the texture brush
            //and then apply a white wash
            g.FillRectangle(backgroundBrush, ClientRectangle);
            g.FillRectangle(new SolidBrush(Color.FromArgb(180, Color.White)), ClientRectangle);

            //Simple draw hello world
            g.DrawString("Hello World", this.Font, new SolidBrush(Color.Black), 10,10);

            //Draw a textured string
            string titleText = "Graphics  Samples";
            g.DrawString(titleText, titleFont, titleShadowBrush, 15, 25);
            g.DrawString(titleText, titleFont, textTextureBrush, 10, 20);

            string textToDraw="Hello Symetrical World";

            //Use Measure string to display a string at the center of the window
            float windowCenter=this.DisplayRectangle.Width/2;
            SizeF stringSize=e.Graphics.MeasureString(textToDraw, textFont);
            float startPos=windowCenter-(stringSize.Width/2);
            g.DrawString(textToDraw, textFont, new SolidBrush(Color.Red), startPos, 10);

            //Now draw a string flowed into a rectangle
            RectangleF rectangle1 = new RectangleF(20, 150, 250, 300);
            g.FillRectangle(new SolidBrush(Color.Gainsboro), rectangle1);
            g.DrawString(flowedText1, textFont, new SolidBrush(Color.Blue), rectangle1);

            //Draw more flowed text but this time center it
            RectangleF rectangle2 = new RectangleF(450, 150, 250, 300);
            g.FillRectangle(new SolidBrush(Color.Gainsboro), rectangle2);
            StringFormat format = new StringFormat();
            format.Alignment=StringAlignment.Center;
            g.DrawString(flowedText2, textFont, new SolidBrush(Color.Blue), rectangle2,format);

            //Work out how many lines and characters we printed just now
            int characters=0;
            int lines=0;
            e.Graphics.MeasureString(flowedText2, textFont, rectangle2.Size, format, out characters, out lines);
            string whatRenderedText=String.Format("We printed {0} characters and {1} lines", characters, lines);
            e.Graphics.DrawString(whatRenderedText, textFont, new SolidBrush(Color.Black), 400,440);

            //If we have the Japanese language pack draw some text in Japanese
            //Rotate it to make life truly exciting
            if (doJapaneseSample) {
                g.RotateTransform(-30);
                g.TranslateTransform(-180, 300);
                g.DrawString(japaneseText, japaneseFont, linearGradBrush, 200, 140);
                g.ResetTransform();
            }

        }

        /// <summary>
        ///    Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
           if (disposing) {
                if (components != null) {
                    components.Dispose();
                }
           }
           base.Dispose(disposing);
        }

        /// <summary>
        ///    Required method for Designer support - do not modify
        ///    the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent() {
            this.components = new System.ComponentModel.Container ();
            this.Size = new System.Drawing.Size(750, 500);
            this.Text = "GDI+ Text Samples";
        }

        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        public static void Main() {
            Application.Run(new TextSample());
        }
    }
}
